In [1]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
from scripts.result_parser import PLOT_PARAMS, get_df

## Free energy of Zika virus

Finally, we present a more challenging problem that studies the free energy of the Zika virus.
We acquired the molecular structure (6CO8) from Protein Data Bank, parameterized it with pdb2pqr and generated mesh on the solvent-excluded surface using Nanoshaper.
The structure contains about 1.6 million atoms (charges) and our mesh has around 10 million boundary elements.
In this experiment, 3 quadrature points were used for regular Galerkin integrals over disjoint elements. The FMM expansion order was 4 and the tolerance of GMRES was $10^{-4}$.

The table below shows the result and performance using both formulations.

In [2]:
direct_6CO8_df = get_df('runs/6CO8/direct/', formulation='direct', skip4=True)
juffer_6CO8_df = get_df('runs/6CO8/juffer/', formulation='juffer', skip4=True)
zika_df = pd.concat([direct_6CO8_df, juffer_6CO8_df])
zika_df.index = ['direct', 'Juffer']
zika_df

t_total_assembly  t_total_gmres  num_iter  e_solv [kcal/Mol]  \
direct       1534.450776    9470.869895       105     -116587.549968   
Juffer       7766.815987    5009.352236        17     -116255.417177   

        memory [GB]  t_fmm_init  t_singular_assembler  t_assemble_sparse  \
direct   109.725492  156.044683            466.436640              25.74   
Juffer   174.958108  151.262922            651.984174              98.07   

        t_assembly_other  t_singular_correction    t_laplace  t_helmholtz  \
direct        886.229454             791.160887  2130.213909  5670.189209   
Juffer       6865.498891             460.615885   776.339616  2796.233184   

        t_avg_laplace  t_avg_helmholtz  t_gmres_other  
direct       4.977135        13.248106     879.305890  
Juffer       5.107497        13.379106     976.163552

-------

### Appendix:

1. results using various regular quadrature order

| quadrature order | expansion order | tol  | e_solv     | gmres time (s) | # iter |
|:----------------:|-----------------|------|------------|----------------|--------|
|         1        | 4               | 1e-4 | -123693.42 | 4149.30        | 105    |
|         2        | 4               | 1e-4 | -116587.55 | 10786.23       | 105    |
|         3        | 4               | 1e-4 | -116491.99 | 13273.08       | 105    |
|         4        | 4               | 1e-4 | -116746.19 | 20891.13       | 105    |

2. result from PyGBe: -117261.125498 kcal/mol